docs/fsharp-cheatsheet.md

```diff
@@ -741,113 +741,165 @@ Another way of implementing interfaces is to use *object expressions*.
+| Async.Sequential        | Composes a new computation from multiple computations, `Async<'T> seq`, and runs them in series; it returns all the results in an array `Async<'T array>`                                                                       |
+| Async.Choice            | Composes a new computation from multiple computations, `Async<'T option> seq`, and returns the first where `'T'` is `Some value` (all others running are canceled). If all computations return `None` then the result is `None` |
+
+For all functions that compose a new computation from children, if any child computations raise an exception, then the overall computation will trigger an exception. All computations that were started will receive a cancellation request.
+
+#### Executing
+
+| Function                     | Description                                                                                                                  |
+|------------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| Async.RunSynchronously       | Runs an async computation and awaits its result.                                                                             |
+| Async.StartAsTask            | Runs an async computation on the ThreadPool and wraps result in a `Task<'T>`.                                                |
+| Async.StartImmediateAsTask   | Runs an async computation, starting immediately on the current operating system thread, and wraps the result in a `Task<'T>` |
+| Async.Start                  | Runs an `Async<unit>` computation on the ThreadPool.                                                                         |
+| Async.StartImmediate         | Runs a computation, starting immediately on the current operating system thread.                                             |
+
+### Cancellation
+
+#### .NET Tasks
+
+.NET Tasks require that you create and share Cancellation Tokens with all sub-Tasks.
+
+    open System
+    open System.Threading
+    open System.Threading.Tasks
+
+    let loop (token: CancellationToken) = task {
+        for cnt in [ 0 .. 10 ] do
+            printf $"{cnt}: And..."
+            do! Task.Delay((TimeSpan.FromSeconds 1), token)  // token is required for Task.Delay to be interrupted
+            printfn "Done"
     }
 
-#### Async.StartAsTask
+    let cts = new CancellationTokenSource (TimeSpan.FromSeconds 3)
+    let runningLoop = loop cts.Token
+    try
+        runningLoop.GetAwaiter().GetResult()  // (1)
+    with :? TaskCanceledException -> printfn "Canceled"
 
-This converts an async computation into a Task. It has the [signature](#functions-signatures): `Async<'T>` -> `Task<'T>`
+Output:
 
-    async {
-        do! Async.Sleep 5000
-    } |> Async.StartAsTask
+    0: And...Done
+    1: And...Done
+    2: And...Canceled
```

Comment:
```suggestion
    1: And...Canceled
```
changed the delay to 2s to emphasize the point that it will be cancelling mid way
3x1s and a 3s timeout is all-but a race condition